data/net/minecraft/data/BlockFamily.mapping

```diff
@@ -0,0 +1,44 @@
+CLASS net/minecraft/data/BlockFamily
+	METHOD <init> (Lnet/minecraft/world/level/block/Block;)V
+		ARG 1 baseBlock
+	METHOD get (Lnet/minecraft/data/BlockFamily$Variant;)Lnet/minecraft/world/level/block/Block;
+		ARG 1 variant
+	CLASS Builder
+		METHOD <init> (Lnet/minecraft/world/level/block/Block;)V
+			ARG 1 block
+		METHOD button (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 button
+		METHOD chiseled (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 chiseled
+		METHOD cracked (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 cracked
+		METHOD cut (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 cut
+		METHOD door (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 door
+		METHOD fence (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 fence
+		METHOD fenceGate (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 fenceGate
+		METHOD polished (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 polished
+		METHOD pressurePlate (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 pressurePlate
+		METHOD recipeGroupPrefix (Ljava/lang/String;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 recipeGroupPrefix
+		METHOD recipeUnlockedBy (Ljava/lang/String;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 recipeUnlockedBy
+		METHOD sign (Lnet/minecraft/world/level/block/Block;Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 sign
+			ARG 2 wallSign
+		METHOD slab (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 slab
+		METHOD stairs (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 stairs
+		METHOD trapdoor (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 trapdoor
+		METHOD wall (Lnet/minecraft/world/level/block/Block;)Lnet/minecraft/data/BlockFamily$Builder;
+			ARG 1 wall
+	CLASS Variant
+		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
+			ARG 3 name
```

Comment:
shouldnt this be ARG 1 ?
@sciwhiz12

Reply:
Nope. For enum constructors, parameter `0` is the instance itself (the constructor acts as an instance method in the eyes of the JVM), parameter `1` is the implicit enum value name, parameter `2` is the ordinal of the enum value (`Enum#ordinal()`), and parameter `3` onwards are the actual declared parameters for the constructor (in this case, the `name` parameter).

Reply:
Okey good to know